src/model/page-config-writer-test.js

```diff
@@ -0,0 +1,255 @@
+
+  beforeEach(() => {
+    win = env.win;
+    doc = win.document;
+    gd = new GlobalDoc(win);
+    markupConfig = {
+      type: 'CreativeWork',
+      isAccessibleForFree: true,
+      isPartOfType: ['CreativeWork', 'Product'],
+      isPartOfProductId: 'scenic-2017.appspot.com:news',
+    };
+  });
+
+  function addJsonLd(content) {
+    const element = createElement(
+      doc,
+      'script',
+      {
+        type: 'application/ld+json',
+      },
+      JSON.stringify(content)
+    );
+    doc.body.appendChild(element);
+    return element;
+  }
+
+  describe('write json-ld', () => {
+    let readyState;
+
+    beforeEach(() => {
+      readyState = 'loading';
+      Object.defineProperty(doc, 'readyState', {get: () => readyState});
+    });
+
+    it('should wait until the element is ready (dom ready)', async () => {
+      const configWriter = new PageConfigWriter(gd);
+      configWriter.writeConfigWhenReady(markupConfig);
+      expect(gd.getHead().childElementCount).to.equal(0);
+    });
+
+    it('should write the config after the doc is read', async () => {
+      readyState = 'complete';
+      const configWriter = new PageConfigWriter(gd);
+      await expect(configWriter.writeConfigWhenReady(markupConfig)).to.be
+        .eventually.fulfilled;
+      expect(gd.getHead().childNodes).to.have.length(1);
+    });
+
+    it('should write a new node if there is no existing schema', async () => {
+      readyState = 'complete';
+      const configWriter = new PageConfigWriter(gd);
+      await expect(configWriter.writeConfigWhenReady(markupConfig)).to.be
+        .eventually.fulfilled;
+      expect(gd.getHead().childNodes).to.have.length(1);
+      expect(gd.getHead().firstChild.textContent).to.equal(
+        JSON.stringify({
+          '@type': 'CreativeWork',
+          'isAccessibleForFree': true,
+          'isPartOf': {
+            'productID': 'scenic-2017.appspot.com:news',
+            '@type': ['CreativeWork', 'Product'],
+          },
+        })
+      );
+    });
+
+    it('should only write a new node once, regardless of the number of calls', async () => {
+      readyState = 'complete';
+      const configWriter = new PageConfigWriter(gd);
+      await expect(configWriter.writeConfigWhenReady(markupConfig)).to.be
+        .eventually.fulfilled;
+      expect(gd.getHead().childNodes).to.have.length(1);
+      expect(gd.getHead().firstChild.textContent).to.equal(
+        JSON.stringify({
+          '@type': 'CreativeWork',
+          'isAccessibleForFree': true,
+          'isPartOf': {
+            'productID': 'scenic-2017.appspot.com:news',
+            '@type': ['CreativeWork', 'Product'],
+          },
+        })
+      );
+      await expect(configWriter.writeConfigWhenReady(markupConfig)).to.be
+        .eventually.fulfilled;
+      expect(gd.getHead().childNodes).to.have.length(1);
+    });
+
+    it('should write a new node if there is an existing but invalid schema', async () => {
```

Comment:
Type should be additive so `@type: "foobar"` would become `@type: [ "foobar", "CreativeWork", "Product" ]` in the json.   Needs to handle both string and array `@type: "foobar"` and `@type: ["foobar"]`

Reply:
Done, I've made both the @type and the isPartOf[@type] fields take either a string or an array of strings, and merged those fields in the PageConfigWriter.